packages/zowe-explorer/src/dataset/actions.ts

```diff
@@ -1439,3 +1481,241 @@ export async function saveFile(doc: vscode.TextDocument, datasetProvider: api.IZ
+        case contextually.isPdsNotFav(node.getParent()):
+            lbl = node.getParent().getLabel().toString() + "(" + node.getLabel().toString() + ")";
+            break;
+        default:
+            vscode.window.showErrorMessage(localize("download.invalidNode", "downloadDataset() called with invalid node."));
+            throw Error(localize("download.invalidNode", "downloadDataset() called with invalid node. "));
+    }
+    const filePath = getDocumentFilePath(lbl, node);
+    return ZoweExplorerApiRegister.getMvsApi(profile).getContents(lbl, {
+        file: filePath,
+        returnEtag: true,
+        encoding: profile.profile.encoding,
+    });
+}
+
+/**
+ * copies given sequential dataset nodes
+ *
+ * @export
+ * @param {ZoweDatasetNode[]} nodes - nodes to be copied
+ */
+export async function copySequentialDatasets(nodes: ZoweDatasetNode[]) {
+    await _copyProcessor(nodes, "ps", async (node: ZoweDatasetNode, dsname: string, replace: shouldReplace) => {
+        const lbl = node.getLabel().toString();
+
+        await api.Gui.withProgress(
+            {
+                location: vscode.ProgressLocation.Window,
+                title: localize("paste.dataSet.InPrg", "Copying File(s)"),
+            },
+            () => {
+                return ZoweExplorerApiRegister.getMvsApi(nodes[0].getProfile()).copyDataSet(lbl, dsname, null, replace === "replace");
+            }
+        );
+    });
+}
+
+/**
+ * copies given partitioned dataset nodes
+ *
+ * @export
+ * @param {ZoweDatasetNode[]} nodes - nodes to be copied
+ */
+export async function copyPartitionedDatasets(nodes: ZoweDatasetNode[]) {
+    await _copyProcessor(nodes, "po", async (node: ZoweDatasetNode, dsname: string, replace: shouldReplace) => {
+        const lbl = node.getLabel().toString();
+        const uploadOptions: IUploadOptions = {
+            etag: node?.getEtag(),
+            returnEtag: true,
+        };
+
+        const children = await node.getChildren();
+        const prof = node?.getProfile();
+        if (prof.profile.encoding) {
+            uploadOptions.encoding = prof.profile.encoding;
+        }
+
+        await api.Gui.withProgress(
+            {
+                location: vscode.ProgressLocation.Window,
+                title: localize("paste.dataSet.InPrg", "Copying File(s)"),
+            },
+            () => {
+                for (const child of children) {
+                    ZoweExplorerApiRegister.getMvsApi(node.getProfile()).copyDataSetMember(
+                        { dsn: lbl, member: child.getLabel().toString() },
+                        { dsn: dsname, member: child.getLabel().toString() },
+                        { replace: replace === "replace" }
+                    );
+                }
+                return Promise.resolve();
+            }
+        );
+    });
+}
+
+/**
+ * Type of z/os dataset or member intended for replacement
+ * @export
+ */
+export type replaceDstype = "ps" | "po" | "da" | "mem";
```

Comment:
I'm not familiar with DA (Direct Access?) data sets. Has this PR been tested to verify we support them?

Reply:
I was trying to understand a bit more what Direct Access (DA) datasets were and, similar to VSAM, these Basic Direct Access Method (BDAM) datasets are a special type that we don't support for copy/paste operations.

I'll remove the DA specific considerations from this PR 😋